### DISCUSSION TITLE
clangd can't resolve cuda feature in .h file

### DISCUSSION THREAD

**Discussion by CalebDu** [clangd/clangd#1684] [Q&A]
i clone a repo [flash attention](https://github.com/HazyResearch/flash-attention). in this repo, build a python package by python setup tool without cmakelist, and many cuda kernel functions ared defined in .h  file. but clangd can't resolve cuda feature correctly in all .h file, such as threadIdx. blockIdx, __syncthread, etc. 
<img width="919" alt="image" src="https://github.com/clangd/clangd/assets/59528230/51078ec1-f4ae-41ee-bd36-74efa94597e3">
i have add some flag in clangd configuration  as following screencut:
<img width="946" alt="image" src="https://github.com/clangd/clangd/assets/59528230/207d7436-0453-4cec-897f-43cd20160c48">
if i rename .h file to .cuh file. the clangd shows "Unable to handle compilation, expected exactly one compiler job in ".
anyone can help me figure it out?

**Comment by HighCommander4** [comment]
Perhaps the tips from other users in https://github.com/clangd/vscode-clangd/issues/125 might be helpful

(Also my comment about `fallbackFlags` in your [other issue](https://github.com/clangd/vscode-clangd/issues/497#issuecomment-1610670034) may be relevant)
